src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,123 @@
-import React from 'react';
+import {
+  ChangeEvent,
+  FC,
+  useCallback,
+  useState,
+} from 'react';
 import './TodoList.scss';
+import { TodoItem } from '../TodoItem/TodoItem';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
+enum Status {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed',
+}
 
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
+type Props = {
+  todos: Todo[];
+  selectedUserId: number,
+  selectNewUser: (x: number) => void;
+};
 
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
+export const TodoList: FC<Props> = ({
+  todos,
+  selectedUserId,
+  selectNewUser,
+}) => {
+  const [filterQuery, setFilterQuery] = useState('');
+  const [status, setStatus] = useState(Status.All);
+  const [shuffle, setShuffle] = useState(false);
+
+  const filteredStatus = () => {
+    switch (status) {
+      case Status.Active:
+        return todos.filter(todo => !todo.completed);
+      case Status.Completed:
+        return todos.filter(todo => todo.completed);
+      case Status.All:
+      default:
+        return todos;
+    }
+  };
+
+  const filteredTodos = filteredStatus().filter(todo => {
+    const title = todo.title.toLowerCase();
+    const query = filterQuery.toLowerCase();
+
+    return title.includes(query);
+  });
 
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
+  const visibleTodos = shuffle
+    ? filteredTodos.sort(() => Math.random() - 0.5)
+    : filteredTodos;
+
+  const changeStatus = useCallback((event: ChangeEvent<HTMLSelectElement>) => {
+    setStatus(event.target.value as Status);
+  }, []);
+
+  const changeFilter = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+    setFilterQuery(event.target.value);
+  }, []);
+
+  const changeShuffle = useCallback(() => {
+    setShuffle(prev => !prev);
+  }, []);
```

Comment:
you should on every click create random orders for todos.
Now every second click returns the orders to default